WalletWasabi/WebClients/BlockstreamInfo/BlockstreamInfoClient.cs

```diff
@@ -0,0 +1,88 @@
+using NBitcoin;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Blockchain.Mempool;
+using WalletWasabi.Tor.Http.Extensions;
+
+namespace WalletWasabi.WebClients.BlockstreamInfo
+{
+	public class BlockstreamInfoClient : IMempoolSyncer
+	{
+		private bool _disposedValue;
+
+		public BlockstreamInfoClient(Network network)
+		{
+			var baseUrl = "https://blockstream.info/";
+			if (network == Network.Main)
+			{
+				baseUrl += "api/";
+			}
+			else if (network == Network.TestNet)
+			{
+				baseUrl += "testnet/api/";
+			}
+			else
+			{
+				throw new NotSupportedException("Specified network not supported.");
+			}
+
+			HttpClient = new HttpClient
+			{
+				BaseAddress = new Uri(baseUrl)
+			};
+			Network = network;
+		}
+
+		public HttpClient HttpClient { get; }
+		public Network Network { get; }
+
+		public async Task<Transaction> GetTransactionAsync(uint256 txid, CancellationToken cancel)
+		{
+			using var response = await HttpClient.GetAsync($"tx/{txid}/hex", cancel).ConfigureAwait(false);
+			if (!response.IsSuccessStatusCode)
+			{
+				await response.ThrowRequestExceptionFromContentAsync().ConfigureAwait(false);
```

Comment:
`ThrowRequestExceptionFromContentAsync` in an interesting function, it looks to me that:

* It is Wasabi Backend specific function because of those content string operations that presumably don't apply to Blockstream API.
* `throw new HttpRequestException($"{me.StatusCode.ToReasonString()}{errorMessage}");` interestingly there is no restriction on the content length so, if I read the code correctly, if you get a large HTTP response, everything is put to the exception message. 
  * One can also use: `throw new HttpRequestException($"{me.StatusCode.ToReasonString()}{errorMessage}", inner: null, me.StatusCode);` (note the last parameter which is available since .NET 5)